request.go

```diff
@@ -225,48 +224,55 @@ func (r *Request) call(handlers Handlers, pkt requestPacket, alloc *allocator, o
 // Additional initialization for Open packets
 func (r *Request) open(h Handlers, pkt requestPacket) responsePacket {
 	flags := r.Pflags()
-	var err error
+
 	switch {
 	case flags.Write, flags.Append, flags.Creat, flags.Trunc:
 		if flags.Read {
-			openFileWriter, ok := h.FilePut.(OpenFileWriter)
-			if ok {
+			if openFileWriter, ok := h.FilePut.(OpenFileWriter); ok {
 				r.Method = "Open"
-				r.state.writerReaderAt, err = openFileWriter.OpenFile(r)
+				rw, err := openFileWriter.OpenFile(r)
+				if err != nil {
+					return statusFromError(pkt, err)
+				}
+				r.state.writerReaderAt = rw
```

Comment:
If we copy-paste the `return &sshFxpHandlePacket{ID: pkt.id(), Handle: r.handle}` here, we don’t need the `if r.Method == "" { … }` anymore. 😄